server/pulp/server/webservices/views/repo_groups.py

```diff
@@ -0,0 +1,370 @@
+                                           kwargs={'repo_group_id': repo_group_id,
+                                                   'distributor_id': distributor['id']})
+        return generate_json_response_with_pulp_encoder(distributor_list)
+
+    @auth_required(authorization.CREATE)
+    @json_body_required
+    def post(self, request, repo_group_id):
+        """
+        Asssociate a distributor with a repo group.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_group_id: matching distributors will be associated with this repo group
+        :type  repo_group_id: str
+
+        :return: response containing a serialized dict of the distributor association
+        :rtype: django.http.HttpResponse
+        """
+        # Params (validation will occur in the manager)
+        params = request.body_as_json
+        distributor_type_id = params.get(distributor_constants.DISTRIBUTOR_TYPE_ID_KEY, None)
+        distributor_config = params.get(distributor_constants.DISTRIBUTOR_CONFIG_KEY, None)
+        distributor_id = params.get(distributor_constants.DISTRIBUTOR_ID_KEY, None)
+        distributor_manager = managers_factory.repo_group_distributor_manager()
+        created = distributor_manager.add_distributor(repo_group_id, distributor_type_id,
+                                                      distributor_config, distributor_id)
+        created['_href'] = reverse('repo_group_distributor_resource',
+                                   kwargs={'repo_group_id': repo_group_id,
+                                           'distributor_id': created['id']})
+        response = generate_json_response_with_pulp_encoder(created)
+        return generate_redirect_response(response, created['_href'])
+
+
+class RepoGroupDistributorResourceView(View):
+    """
+    View for a single repo group distributor association.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, repo_group_id, distributor_id):
+        """
+        Get information about a single repo group distributor association.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_group_id: repo group the distributor is associated with
+        :type  repo_group_id: str
+        :param distributor_id: distributor to get
+        :type  distributor_id: str
+
+        :return: response containing a serialized dict of the added distributor
+        :rtype: django.http.HttpResponse
+        """
+        distributor_manager = managers_factory.repo_group_distributor_manager()
+        dist = distributor_manager.get_distributor(repo_group_id, distributor_id)
+        dist['_href'] = request.get_full_path()
+        return generate_json_response_with_pulp_encoder(dist)
+
+    @auth_required(authorization.DELETE)
+    @json_body_allow_empty
```

Comment:
By allowing empty, a body is not necessary and will follow the default path of `force=false`. You could also send a DELETE request with the body `{"force": true}`.